haas/ext/switches/_console.py

```diff
@@ -65,26 +66,29 @@ def disable_native(self, vlan_id):
         """
 
     @abstractmethod
+    def disable_port(self):
+        """Disable all vlans on the current port."""
+
+    @abstractmethod
     def disconnect(self):
         """End the session. Must be at the main prompt."""
 
-    def apply_networking(self, action):
-        interface = action.nic.port.label
-        channel = action.channel
+    def modify_port(self, port, channel, network_id):
+        interface = port
+        port = Port.query.filter_by(label=port,
+                                    owner_id=self.switch.id).one()
 
         self.enter_if_prompt(interface)
         self.console.expect(self.if_prompt)
 
         if channel == 'vlan/native':
-            old_native = None
-            old_attachments = filter(lambda a: a.channel == 'vlan/native',
-                                     action.nic.attachments)
-            if len(old_attachments) != 0:
-                old_native = old_attachments[0].network.network_id
-            if action.new_network is None:
+            old_native = NetworkAttachment.query.filter_by(
+                channel='vlan/native',
```

Comment:
just a reminder: you need to merge master back into this to get these changes
https://github.com/CCI-MOC/hil/blob/master/haas/ext/switches/_console.py#L80

without this the deployment tests for cisco and dell fail.